source/JsMaterialX/README.md

```diff
@@ -1,105 +1,68 @@
-
-  3. Create a build directory.
-     ```sh
-     docker exec -it emscripten sh -c "[ -d build ] || mkdir build"
-     ``` 
-  4. Build the JavaScript bindings.
-     ```sh
-     docker exec -it emscripten sh -c "cd build && cmake .. -DMATERIALX_BUILD_JS=ON -DMATERIALX_BUILD_RENDER=OFF -DMATERIALX_BUILD_TESTS=OFF -DMATERIALX_EMSDK_PATH=/emsdk_portable/ && cmake --build . --target install"
-     ```
-#### CMake
-The JavaScript library can be built using cmake and make.
-
-1. Create a build folder from in the *root* of the repository.
+Create a build folder in the *root* of the repository and navigate to that folder:
 ```sh
-mkdir -p ./build
+mkdir ./build
 cd ./build
 ```
 
-2. Run cmake and make
-When building the JavaScript output the user can specify the emsdk path with the `MATERIALX_EMSDK_PATH` option.
-This option can be omitted if the `emsdk/emsdk_env.sh` script was run beforehand.
+If you are using the emsdk directly on Windows, note that the emscripten SDK doesn't work with Microsoft's Visual Studio build tools. You need to use an alternative CMake generator like [MinGW](http://mingw-w64.org/doku.php) Makefiles or [Ninja](https://ninja-build.org/). We recommend to use Ninja (unless you already have MinGW installed), since it's pretty lightweight and a pure build system, instead of a full compiler suite.
 
+Generate the build files with CMake. When building the JavaScript bindings, you can optionally specify the emsdk path with the `MATERIALX_EMSDK_PATH` option. This option can be omitted if the `emsdk/emsdk_env.sh` script was run beforehand.
 ```sh
-# This will generate the release library
-cmake .. -DMATERIALX_BUILD_JS=ON -DMATERIALX_BUILD_RENDER=OFF -DMATERIALX_BUILD_TESTS=OFF -DMATERIALX_EMSDK_PATH=/mnt/c/GitHub/PUBLIC/emsdk
-cmake --build .
+cmake .. -DMATERIALX_BUILD_JS=ON -DMATERIALX_BUILD_RENDER=OFF -DMATERIALX_BUILD_TESTS=OFF -DMATERIALX_BUILD_GEN_GLSL=OFF -DMATERIALX_BUILD_GEN_OSL=OFF -DMATERIALX_BUILD_GEN_MDL=OFF -DMATERIALX_BUILD_GEN_OGSXML=OFF -DMATERIALX_BUILD_GEN_OGSFX=OFF -DMATERIALX_BUILD_GEN_ARNOLD=OFF -DMATERIALX_BUILD_VIEWER=OFF -DMATERIALX_EMSDK_PATH=/path/to/emsdk
 ```
+On Windows, remember to set the CMake generator via the `-G` flag , e.g. `-G "Ninja"`.
 
-For Windows use [Ninja](https://ninja-build.org/) as the cmake generator.
+To build the project, run
+```sh
+cmake --build . --target install -j8
+```
+Change the value of the `-j` option to the number of threads you want to build with.
+#### Docker
+In order to build using the Docker image, execute the following command in a terminal (in the root of this repository, after creating the build folder), using the CMake commands introduced above.
 ```sh
-cmake .. -DMATERIALX_BUILD_JS=ON -DMATERIALX_BUILD_RENDER=OFF -DMATERIALX_BUILD_TESTS=OFF -DMATERIALX_EMSDK_PATH=C:\GitHub\PUBLIC\emsdk -G "Ninja"
-cmake --build .
+docker run --rm -v {path_to_MaterialX_repo}:/src emscripten/emsdk sh -c "cd build && <cmake generator command> && <cmake build command>"
```

Comment:
just personal flavor since I skim through readmes looking for the most important info. Can we link cmake generator command to the command above

Reply:
I tried to do that, but I couldn't find a way to link from within a code block.